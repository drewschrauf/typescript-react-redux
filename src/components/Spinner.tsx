import React from 'react';

const Spinner = () => <div className="spinner" />;
export default Spinner;
