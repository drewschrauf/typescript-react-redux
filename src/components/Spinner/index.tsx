import React from 'react';
import './styles.css';
import withDelay from '../../hoc/withDelay';

const Spinner = () => <div className="spinner" />;

export default withDelay({ delay: 500 })(Spinner);
