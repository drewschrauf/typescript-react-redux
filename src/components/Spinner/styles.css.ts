import { keyframes, style } from '@vanilla-extract/css';

const spin = keyframes({
  '0%': {
    transform: 'rotate(0deg)',
  },
  '100%': {
    transform: 'rotate(360deg)',
  },
});

export const spinner = style({
  width: '64px',
  height: '64px',
  margin: '0 auto',

  ':after': {
    content: ' ',
    display: 'block',
    width: '46px',
    height: '46px',
    margin: '1px',
    borderRadius: '50%',
    border: '5px solid black',
    borderColor: 'black transparent black transparent',
    animation: `${spin} 1.2s linear infinite`,
  },
});
