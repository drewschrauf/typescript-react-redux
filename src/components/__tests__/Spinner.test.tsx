import React from 'react';
import { mount } from 'enzyme';

import Spinner from '../Spinner';

describe('Spinner', () => {
  it('should not render anything on initial render', () => {
    const root = mount(<Spinner />);
    expect(root.html()).toBe(null);
  });

  it('should render spinner after a timeout', () => {
    jest.useFakeTimers();
    const root = mount(<Spinner />);
    jest.runAllTimers();
    root.update();
    expect(root.html()).toMatchSnapshot();
  });
});
