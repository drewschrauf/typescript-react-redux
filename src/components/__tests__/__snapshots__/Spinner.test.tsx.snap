// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render spinner after a timeout 1`] = `"<div class=\\"spinner\\"></div>"`;
